cmd/soroban-cli/src/config/mod.rs

```diff
@@ -77,15 +75,24 @@ pub struct Args {
 
 impl Args {
     pub fn key_pair(&self) -> Result<ed25519_dalek::Keypair, Error> {
-        let key = if let Some(identity) = &self.identity {
-            locator::read_identity(identity)?
-        } else if let Some(secret_key) = &self.secret_key {
-            secret::Secret::SecretKey {
-                secret_key: secret_key.clone(),
+        let key = if let Some(source_account) = &self.source_account {
+            let parts = source_account
+                .split_once(':')
+                .ok_or_else(|| Error::MalformedSourceAccount(source_account.clone()))?;
+            match parts {
+                ("id", identity) => locator::read_identity(identity)?,
+                ("sk", secret_key) => secret::Secret::SecretKey {
+                    secret_key: secret_key.to_string(),
+                },
+                ("seed", seed_phrase) => secret::Secret::SeedPhrase {
+                    seed_phrase: seed_phrase.to_string(),
+                },
+                _ => return Err(Error::MalformedSourceAccount(source_account.clone())),
```

Comment:
I was just going off the initial proposal of `id:..` so it was just easy to add prefixes to the other ones.

I can try for a method that detects it from context, but it would also be nice to embed an hd path as part of the arg.

e.g. `id:test:2`, but this could be done without the prefix, but is more explicit.